packages/plugins/documentation/__tests__/build-component-schema.test.js

```diff
@@ -0,0 +1,321 @@
+'use strict';
+
+const buildComponentSchema = require('../server/services/helpers/build-component-schema');
+const strapi = require('../__mocks__/strapi');
+
+describe('Build Component Schema', () => {
+  it('builds the Response schema', () => {
+    global.strapi = strapi;
+
+    const apiMocks = [
+      {
+        name: 'users-permissions',
+        getter: 'plugin',
+        ctNames: ['role'],
+      },
+      { name: 'restaurant', getter: 'api', ctNames: ['restaurant'] },
+    ];
+
+    let schemas;
+    for (const mock of apiMocks) {
+      schemas = {
+        ...schemas,
+        ...buildComponentSchema(mock),
+      };
+    }
+
+    const schemaLength = Object.keys(schemas).length;
```

Comment:
I understand that doing all in one Object.entries wouldn't be very readable, but maybe we could store `Object.keys(schemas)` in a variable there

Reply:
oh yeah I thought I had changed that everywhere but I must have missed one